ci-scripts/test_agent/FirebaseClient.py

```diff
@@ -34,14 +34,16 @@ def __init__(self):
         self.db = self.firebase.database()
 
     def push_test_report(self, workload, verdict, report):
+        build_id = workload.val().get("build_id")
         data = {
             "timestamp": int(time.time()),
             "verdict": verdict,
             "workload": workload.val(),
             "report": report,
         }
-        self.db.child("workers").child(self.config["agent_id"]).child("reports").push(
-            data, self.user["idToken"]
+        # Use build_id as key for report. Overwrites any existing report
+        self.db.child("workers").child(self.config["agent_id"]).child("reports").child(build_id).set(
+            data, self.user["idToken"],
```

Comment:
Can you rebase. Master already has this.

Reply:
i did and just pushed. this should be resolved now